.github/workflows/ci_release.yml

```diff
@@ -50,19 +50,48 @@ jobs:
   go-ci:
     uses: ./.github/workflows/go-ci.yml
 
-  # Make a release if this is a manually trigger job, i.e. workflow_dispatch
-  release:
+  # If this was a workflow dispatch event, we need to generate and push a tag
+  # for goreleaser to grab
+  version_bump:
     needs: [hadolint, yamllint, markdown-lint, go-ci]
     runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'workflow_dispatch' }}
     permissions: "write-all"
     steps:
       - uses: actions/checkout@v3
-      - name: Version Release
-        uses: celestiaorg/.github/.github/actions/version-release@v0.2.2
+      - name: Bump version and push tag
+        # Placing the if condition here is a workaround for needing to block
+        # on this step during workflow dispatch events but the step not
+        # needing to run on tags. If we had the if condition on the full
+        # version_bump section, it would skip and not run, which would result
+        # in goreleaser not running either.
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        uses: mathieudutour/github-tag-action@v6.0
         with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          version-bump: ${{inputs.version}}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          default_bump: ${{ inputs.version }}
+
+  # Generate the release with goreleaser to include pre-built binaries
+  goreleaser:
+    needs: version_bump
+    runs-on: ubuntu-latest
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'push' && contains(github.ref, 'refs/tags/'))
+    permissions: "write-all"
+    steps:
+      - uses: actions/checkout@v3
+      - run: git fetch --force --tags
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
```

Comment:
Is it possible to make this var? We have it in like 3-5 places in CI and changing it manually all the time is a PITA

Reply:
yea there are a number of ways you could make it a var. Happy to help in a follow up issue to clean up other places in the CI 👍